airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/sponsored_brands.py

```diff
@@ -12,12 +12,23 @@ class SponsoredBrandsCampaigns(SubProfilesStream):
     https://advertising.amazon.com/API/docs/en-us/sponsored-brands/3-0/openapi#/Campaigns
     """
 
+    def __init__(self, *args, **kwargs):
```

Comment:
The use of `state_filter` parameter is copy-pasted across all three streams. Refactor it to a mixin instead, e.g.

```python
class StateFilterMixin:
    state_filter = None

    def __init__(self, *args, **kwargs):
        self.state_filter = kwargs.get("config", {}).get("state_filter")
        super().__init__(*args, **kwargs)

    def request_params(self, *args, **kwargs):
        params = super().request_params(*args, **kwargs)
        if self.state_filter:
            params["stateFilter"] = ",".join(self.state_filter)
        return params

class SponsoredBrandsCampaigns(StateFilterMixin, SubProfilesStream):
    ...
```

Place `StateFilterMixin` in `common.py`.